src/SmartComponents/modules/reducers.js

```diff
@@ -234,13 +245,28 @@ function toggleExpandedRow(expandedRows, factName) {
     return expandedRows;
 }
 
+function updateStateFilters(stateFilters, updatedStateFilter) {
+    let newStateFilters = [];
+
+    stateFilters.forEach(function (stateFilter) {
+        if (stateFilter.filter === updatedStateFilter.filter) {
+            newStateFilters.push(updatedStateFilter);
+        } else {
+            newStateFilters.push(stateFilter);
+        }
+    });
```

Comment:
I'm surprised by this logic. Based on your data structure, I was assuming that updating the set of filters would be setting 'selected' to true or false based on whether a user selected or unselected a filter.

Reply:
So, this is annoying. If I change the function like so:
```
    stateFilters.forEach(function (stateFilter) {
        if (stateFilter.filter === updatedStateFilter.filter) {
            stateFilter.selected = updatedStateFilter.selected;
        }
    });

    return stateFilters;
```
the checkboxes don't check and uncheck until I close the dropdown box.